test/e2e/framework/service/jig.go

```diff
@@ -355,9 +355,10 @@ func (j *TestJig) WaitForEndpointOnNode(ctx context.Context, nodeName string) er
 
 // waitForAvailableEndpoint waits for at least 1 endpoint to be available till timeout
 func (j *TestJig) waitForAvailableEndpoint(ctx context.Context, timeout time.Duration) error {
+	internalCtx, cancel := context.WithTimeout(ctx, timeout)
```

Comment:
I would replace `ctx`:
```
ctx, cancel := context.WithTimeout(ctx, timeout)
defer cancel()
```

Then you don't need update the rest of the code, except for `wait.PollUntilContextTimeout` which can be replaced by `wait.PollUntilWithContext`.

The advantage is that it becomes impossible to accidentally use the `ctx` without the timeout.

Reply:
👍

Reply:
done